pkg/performanceprofile/utils/schedstat/schedstat.go

```diff
@@ -0,0 +1,88 @@
+package schedstat
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	DefaultPath = "/proc/schedstat"
+)
+
+func MakeCPUIDListFromCPUList(cpus []string) ([]int, error) {
+	cpuIDs := make([]int, 0, len(cpus))
+	for _, cpu := range cpus {
+		if !strings.HasPrefix(cpu, "cpu") {
+			continue
+		}
+		// len("cpu") == 3
+		cpuID, err := strconv.Atoi(cpu[3:])
+		if err != nil {
+			return cpuIDs, err
+		}
+		cpuIDs = append(cpuIDs, cpuID)
+	}
+	return cpuIDs, nil
+}
+
+type Info struct {
+	cpuDomains map[string][]string
+}
+
+func (inf Info) GetCPUs() []string {
+	ret := []string{}
+	for cpu := range inf.cpuDomains {
+		ret = append(ret, cpu)
+	}
+	sort.Strings(ret)
+	return ret
+}
+
+func (inf Info) GetDomainsByID(cpuID int) ([]string, bool) {
+	return inf.GetDomains(fmt.Sprintf("cpu%d", cpuID))
+}
+
+func (inf Info) GetDomains(cpu string) ([]string, bool) {
+	doms, ok := inf.cpuDomains[cpu]
+	return doms, ok
+}
+
+func ParseData(r io.Reader) (Info, error) {
```

Comment:
I wonder if we should add a comment on this function expected input/output for future use (or code is enough by itself)